crates/runtime/src/tool_use.rs

```diff
@@ -106,7 +108,100 @@ pub trait SpiceModelTool: Sync + Send {
     ) -> Result<Value, Box<dyn std::error::Error + Send + Sync>>;
 }
 
-pub struct SqlTool {}
+pub struct DocumentSimilarityTool {}
+
+#[derive(Deserialize)]
+struct DocumentSimilarityToolArgs {
+    text: String,
+    from: Vec<String>,
+    limit: Option<usize>,
+}
+
+#[async_trait]
+impl SpiceModelTool for DocumentSimilarityTool {
+    fn name(&self) -> String {
```

Comment:
```suggestion
    fn name(&self) -> &'static str {
```

This isn't ever generated at runtime right?

Reply:
Not yet, so happy to change